### PR TITLE
ebooks: fix ebooks dojson

### PR DESCRIPTION
**Pull request by rerowep**
* Fixes documents dumps for records without series.
    * Corrects series in dojson from ebooks.
    * Adds "_text" to json schema.
    * Adds `invenio utils reindex` and `invenio utils runindex` commands
      for bulk indexing with custom class.

Co-Authored-by: Peter Weber <peter.weber@rero.ch>

## Why are you opening this PR?

- ebooks api and json chrashed

## How to test?

- import ebooks and display the json or api of an ebbok without "series"

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1.json [hunk 1]
```diff
@@ -1613,6 +1613,10 @@
             "items": {
               "type": "object",
               "properties": {
+                "_text": {
+                  "title": "Calculated text",
```
Comment (jma) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1.json]: Can you change this such as: normalized version of the provisionActivity or provisionActivity statement, etc.

**Comment by jma** [comment]
It seems that some code are missing, such as jsonschema tests, elasticsearch mapping, _text properties assignment in record dumps, or indexing. Is it a part of an other PR?

File: rero_ils/modules/api.py [hunk 2]
```diff
@@ -85,6 +86,80 @@ def delete(self, record):
...
+        :param record: The record to prepare.
+        :param index: The Elasticsearch index.
+        :param doc_type: The Elasticsearch document type.
+        :param arguments: The arguments to send to Elasticsearch upon indexing.
+        :param **kwargs: Extra parameters.
+        :returns: The record metadata.
+        """
+        import pytz
+        if current_app.config['INDEXER_REPLACE_REFS']:
+            data = deepcopy(record.replace_refs().dumps())
```
Comment (jma) [inline rero_ils/modules/api.py]: deepcopy in not required: https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/api.py#L134

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json [hunk 3]
```diff
@@ -1615,6 +1615,10 @@
             "items": {
               "type": "object",
               "properties": {
+                "_text": {
```
Comment (jma) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json]: should not be here. Only in the ES mapping

File: rero_ils/modules/utils.py [hunk 4]
```diff
@@ -37,16 +37,16 @@ def strtotime(strtime):
     )
 
 
-def bulk_index(uuids, process=False, verbose=False):
+def do_bulk_index(uuids, doc_type=None, process=False, verbose=False):
```
Comment (jma) [inline rero_ils/modules/utils.py]: why did you change the name?

File: scripts/setup [hunk 5]
```diff
@@ -212,18 +212,18 @@ fi
...
+pipenv run invenio utils reindex -t item --yes-i-know
+pipenv run invenio utils runindex -c 4 --raise-on-error
 
 display_success_message "Index Documents:"
-pipenv run invenio index reindex -t doc --yes-i-know
-pipenv run invenio index run -c 4 --raise-on-error
+pipenv run invenio utils reindex -t doc --yes-i-know
+pipenv run invenio utils runindex -c 4 --raise-on-error
 
 set +e
```
Comment (jma) [inline scripts/setup]: please remove this

File: rero_ils/modules/api.py (hunk 2 above)
Comment (rerowep) [reply rero_ils/modules/api.py]: ok

File: rero_ils/modules/utils.py (hunk 4 above)
Comment (rerowep) [reply rero_ils/modules/utils.py]: Because it is not the bulk_index from the class but a function with bulk_index and reindex.

File: rero_ils/modules/api.py [hunk 6]
```diff
@@ -85,6 +86,80 @@ def delete(self, record):
...
+        """Prepare record data for indexing.
+
+        :param record: The record to prepare.
+        :param index: The Elasticsearch index.
+        :param doc_type: The Elasticsearch document type.
+        :param arguments: The arguments to send to Elasticsearch upon indexing.
+        :param **kwargs: Extra parameters.
+        :returns: The record metadata.
+        """
+        import pytz
```
Comment (BadrAly) [inline rero_ils/modules/api.py]: move the import to the top of file if possible.

File: rero_ils/modules/api.py [hunk 7]
```diff
@@ -85,6 +86,80 @@ def delete(self, record):
...
+        else:
+            data = record.dumps()
+
+        data['_created'] = pytz.utc.localize(record.created).isoformat() \
+            if record.created else None
+        data['_updated'] = pytz.utc.localize(record.updated).isoformat() \
+            if record.updated else None
+
+        # Allow modification of data prior to sending to Elasticsearch.
+        from invenio_indexer.signals import before_record_index
```
Comment (BadrAly) [inline rero_ils/modules/api.py]: same as above

File: rero_ils/modules/cli.py [hunk 8]
```diff
@@ -861,3 +869,64 @@ def reserve_pid_range(pid_type, records_number, unused):
                                              status=PIDStatus.NEW)
                 db.session.add(identifier(recid=pid))
             db.session.commit()
+
+
+@utils.command('runindex')
```
Comment (BadrAly) [inline rero_ils/modules/cli.py]: How about just use `bulkindex` instead of `runindex`

File: rero_ils/modules/api.py (hunk 6 above)
Comment (rerowep) [reply rero_ils/modules/api.py]: Correct, but was not from me!

File: rero_ils/modules/cli.py (hunk 8 above)
Comment (rerowep) [reply rero_ils/modules/cli.py]: Because this is the command `run` the index -> invenio indexer `invenio index run`
